### ISSUE TITLE
Bug in continue follow the leader GP

### ISSUE THREAD

**Issue by deveee** [supertuxkart/stk-code#2349]
As reported in the forum:
> If you save a Grand prix halfway, when you restart you play the leader instead of a racer. Funny though...!

This probably occurs when you won previous races and you are first in the rank.

**Comment by deveee** [comment]
@konstin could you explain why you sort karts in saved_grand_prix.cpp in loadKarts() ?

I mean this commit:
035fc599ee1485be90eb429ead581142138112eb
and this line:
std::sort(kart_list.begin(), kart_list.end(), cmp__l);

This causes an issue that the leader is not leader anymore when you continue GP.

After short testing, the karts seem to be in proper order, even without sorting.

**Comment by konstin** [comment]
iirc because should be start in the order they ranked and this doesn't always work without sorting. But tbh, I'm don't know anymore why I did that.

**Comment by deveee** [comment]
I don't remember too :P

But as I see it is sorted in startNextRace anyway. And it leaves the leader at first place:
```
    if (m_track_number > 0)
    {
        // In follow the leader mode do not change the first kart,
        // since it's always the leader.
        int offset = (m_minor_mode==MINOR_MODE_FOLLOW_LEADER) ? 1 : 0;

        // Keep players at the end if needed
        int player_last_offset = 0;
        if (UserConfigParams::m_gp_player_last)
        {
            // Doing this is enough to keep player karts at
            // the end because of the simple reason that they
            // are at the end when getting added. Keep them out
            // of the later sorting and they will stay there.
            player_last_offset = (int)m_player_karts.size();
        }

        std::sort(m_kart_status.begin()+offset,
                  m_kart_status.end() - player_last_offset);
```

I will leave it commented for now.

**Comment by deveee** [comment]
Fixed in 054d9251bba40926613ada4330c2217bee042a02
